src/Adornment/AdornmentContainer.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { FC, Dispatch, useRef, useEffect } from 'react'
+
+import { StyledAdornmentProps, StyledAdornment } from './styled'
+
+interface AdornmentContainer extends StyledAdornmentProps {
+  registerWidth: Dispatch<number>
+}
+
+export const AdornmentContainer: FC<AdornmentContainer> = ({ registerWidth, children, position, ...rest }) => {
```

Comment:
Any suggestions for this name as well? This is the component wherein the actual Adornment gets rendered

Reply:
Ill think about it.

I do have other suggestion - how about removing registerWidth and instead forwarding a ref (preferable) or make use of something like an innerRef prop. This way we can isolate the adjusting width logic in the wrapper component.